### PR TITLE
derive FileConfig

### PR DESCRIPTION
**Pull request by ModProg**

### CODE REVIEW

File: crates/bonsaidb-macros/src/lib.rs [hunk 1]
```diff
@@ -723,6 +727,134 @@ pub fn api_derive(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
...
+    let core = core.unwrap_or_else(core_path);
+    let files = files.unwrap_or_else(files_path);
+    let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
+
+    // TODO error if only authority is given
+    // let (files_name, blocks_name) = if let (Some (authority) = authority {
+    //     let files_name = files_name.unwrap_or_else(|| abort_call_site!("if `authority` is given, `files_name` and `blocks_name need to be provided as well"));
+    //     (,())
+    // }
+
```
Comment (ecton) [inline crates/bonsaidb-macros/src/lib.rs:806]:
```suggestion
```

File: crates/bonsaidb-macros/src/lib.rs [hunk 2]
```diff
@@ -723,6 +727,134 @@ pub fn api_derive(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
...
+    let (files_name, blocks_name) = match (authority, files_name, blocks_name) {
+        (None, None, None) => (
+            quote!(#files::BonsaiFiles::files_name()),
+            quote!(#files::BonsaiFiles::blocks_name()),
+        ),
+        (Some(authority), Some(files_name), Some(blocks_name)) => (
+            quote!(#core::schema::Qualified::new(#authority, #files_name)),
```
Comment (ecton) [inline crates/bonsaidb-macros/src/lib.rs:813]: Huh, invoking `Qualified::new` works here? I think some of the other code has a more complex `as` expression that could be simplified. That doesn't need to be done in this PR, just so astounded that this works that I'm leaving the comment :)

Neat!

**Review by ecton — APPROVED** [review]
Looks great!

Due to some issues in main, I'm going to hold off on this merge until I've merged #250.

File: crates/bonsaidb-macros/src/lib.rs (hunk 1 above)
Comment (ModProg) [reply crates/bonsaidb-macros/src/lib.rs:806]: Ups, must have forgotten to remove that when introducing the match :D

**Pull request merged**